File: python/anagram/anagram.py
def find_anagrams(word, candidates):
    word_lower = word.lower()
    ordered_word = sorted(list(word_lower))
    anagrams = []
    for candidate in candidates:
        candidate_lower = candidate.lower()
        if sorted(list(candidate_lower)) == ordered_word and candidate_lower != word_lower:
            anagrams.append(candidate)
    return anagrams
        

File: elixir/anagram/lib/anagram.ex
defmodule Anagram do
  @doc """
  Returns all candidates that are anagrams of, but not equal to, 'base'.
  """
  @spec match(String.t(), [String.t()]) :: [String.t()]
  def match(base, candidates) do
    base_letters = count_letters(String.downcase(base))
    Enum.filter(candidates, fn (candidate) -> String.downcase(base) != String.downcase(candidate) and base_letters == count_letters(String.downcase(candidate)) end)
  end

  defp count_letters(word) do
    Enum.reduce(String.graphemes(word), %{}, fn char, acc -> Map.put(acc, char, (acc[char] || 0) + 1) end)
  end
end
